<?xml version="1.0"?>
<launch>
    <node pkg="rosbag" type="play" name="rosbag_play" args="--clock -r 1 /home/jeff/Downloads/dataset/Imu_Depth_Color_Points.bag"/>
    <param name="/use_sim_time" value="true" />

    <node pkg="tf" type="static_transform_publisher" name="word2map_tf"  args="0 0 0 1.57 3.14 1.57 /world /map 100" />
    <node pkg="tf" type="static_transform_publisher" name="base2imu_tf"  args="0 0 0 0 0 0 /odom /camera_depth_optical_frame 100" />
<!-- 
    <include file="$(find ssl_slam3)/launch/rs_camera.launch">
        <arg name="color_width" value="1280" />
        <arg name="color_height" value="720" />
        <arg name="filters" value="pointcloud" />    
    </include>
     -->
    <param name="file_path" value="$(find ssl_slam3)/config/params.yaml"/>

    <node pkg="ssl_slam3" type="ssl_slam3_laser_processing_node" name="ssl_slam3_laser_processing_node" output="screen" />
    <node pkg="ssl_slam3" type="ssl_slam3_odom_estimation_node" name="ssl_slam3_odom_estimation_node" output="screen" />
    <node pkg="ssl_slam3" type="ssl_slam3_laser_mapping_node" name="ssl_slam3_laser_mapping_node" output="screen"/>

    <arg name="rviz" default="true" />
    <group if="$(arg rviz)">
        <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find ssl_slam3)/rviz/ssl_slam3.rviz" />
    </group>

    <node pkg="hector_trajectory_server" type="hector_trajectory_server" name="trajectory_server_loam" ns="ssl_slam3" >
        <param name="/target_frame_name" value="map" />
        <param name="/source_frame_name" value="odom" />
        <param name="/trajectory_update_rate" value="10.0" />
        <param name="/trajectory_publish_rate" value="10.0" />
    </node>

</launch>
